editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/restconfiguration/RestEditorTestIT.java

```diff
@@ -199,6 +199,90 @@ public void openCamelFileAddRestElementTest() throws Exception {
 		assertThat(re.getId()).isNotEmpty();
 	}
 
+	@Test
+	public void openCamelFileRestElementIdChangeAndRemoveOperationTest() throws Exception {
+		// FUSETOOLS-3088
+		IEditorPart openEditorOnFileStore = openFileInEditor(NON_REST_PROJECT);
+
+		readAndDispatch(20);
+
+		// ensure that the REST tab is available and select it
+		CamelEditor camelEditor = (CamelEditor)openEditorOnFileStore;
+		camelEditor.setActiveEditor(camelEditor.getRestEditor());
+		RestConfigEditor restEditor = camelEditor.getRestEditor();
+		camelEditor.setActiveEditor(restEditor);
+		
+		// add a REST element
+		new AddRestElementAction(restEditor, null).run();
+		readAndDispatch(20);
+
+		StructuredSelection ssel = (StructuredSelection) restEditor.getSelection();
+		RestElement re = (RestElement) ssel.getFirstElement();
+		String newId = "changed-id";
+		re.setId(newId);
+		assertThat(re.getId().equals(newId));
+		
+		Display.getDefault().syncExec(() -> 
+			restEditor.createRestOperation(RestVerbElement.GET_VERB, "test", "myTestID", "")
+		);
+		readAndDispatch(20);
+
+		Display.getDefault().syncExec(() -> 
+			restEditor.createRestOperation(RestVerbElement.GET_VERB, "test2", "myTestID2", "")
+		);
+		readAndDispatch(20);
+		
+		RestVerbElement restVerbElement = 
+				(RestVerbElement) re.getRestOperations().values().iterator().next();
+		assertThat(restVerbElement.getId().equals("myTestID"));
+		restEditor.selectRestVerbElement(restVerbElement);
+		
+		//test to make sure it's removed (there are two REST elements added)
+		new DeleteRestOperationAction(restEditor, null).run();
+		readAndDispatch(20);
+		assertThat(re.getRestOperations().size() == 1);
+		RestVerbElement restVerbElement2 = 
+				(RestVerbElement) re.getRestOperations().values().iterator().next();
+		assertThat(restVerbElement2.getId().equals("myTestID2"));
```

Comment:
I think it should be assertThat(restVerbElement2.getId()).isEqualTo("myTestID2");